contracts/VolOracle.sol

```diff
@@ -102,4 +102,67 @@ contract VolOracle {
     function getObservation(address _pool, uint256 _idx) public view returns (VolObservation memory) {
         return oracleStates[_pool].observations[_idx];
     }
+
+
+    // @dev get the standdevation given specified days range from now
+    function getVol(address _pool, uint32 _daysToNow) public view returns(uint256 standardDeviation) {
+        uint32 target = uint32(block.timestamp) - _daysToNow * uint32(1 days);
+        // binary search to start index.
+        uint256 startIndex = getObservationIndexBeforeOrAtTarget(_pool, target);
+        // TODO: edge case - 1st element of the array
+
+        return calculateVol(_pool, startIndex);
+    }
+
+    // TODO: WIP
+    // @dev calculate the standdevation from startIndex to endIndex
+    function calculateVol(address _pool, uint256 startIndex) internal view returns(uint256 stanDeviation) {
+        VolOracleState memory volOracleState = oracleStates[_pool];
+        uint256 endIndex = volOracleState.observationIndex;
+        VolObservation memory startObservation = volOracleState.observations[startIndex];
+        VolObservation memory endObservation = volOracleState.observations[endIndex];
+        uint256 tickSquareSum = endObservation.tickSquareCumulative - startObservation.tickSquareCumulative;
+        uint256 timeEclapsed = uint256(endObservation.blockTimestamp - startObservation.blockTimestamp);
+        uint256 tickAvg = uint256(uint56(endObservation.tickCumulative - startObservation.tickCumulative)) / timeEclapsed;
+        uint256 stddev = (tickSquareSum - tickAvg * tickAvg * timeEclapsed) / timeEclapsed;
```

Comment:
this is correct

Reply:
https://www.notion.so/Writeup-technical-details-of-Welford-current-stddev-fd367ff58b0b4ede9e8ba112c64316dd#c272137a5c714ecdb0c660cfadcc2cd2